src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+#
+#Variable Step Variable Order (VSVO) Adams-Moulton Solver for nonstiff problems
+#
+#(Main ref: Hairer & Wanner 1996, Vol I, p.356-360, p.398-400, p.421-423)
+"""
+function ode113(F, y0, tspan, order::Integer = 4; kwords...)
+    # For y0 which don't support indexing.
+    F_ = (t, y) -> [F(t, y[1])]
+    t, y = ode113(F_, [y0], tspan, order; kwords...)
+    return t, vcat_nosplat(y)
+end
+
+"a variable stepsize, variable order multi-step solver for nonstiff ODEs"
+function ode113(F, y0::AbstractVector, tspan, order::Integer = 4;
+                                        adaptive_order = true,
+                                        max_order = 13,
+                                        reltol = 1.0e-9,
+                                        abstol = 1.0e-9,
+                                        minstep = abs(tspan[1] - tspan[end]) / 1e18,
+                                        maxstep = abs(tspan[end] - tspan[1]) / 2.5,
+                                        initstep = 0.,
+                                        fac = .8,
+                                        norm = Base.norm,
+                                        points = :all)
+
+    #-##########################################################################
+    # Initializations
+    #-##########################################################################
+    Et, Eyf, Ty = make_consistent_types(F, y0, tspan)
+    dof = length(y0)
+
+    # initialization of time span, which will have variable length
+    tspan = convert(Vector{Et}, tspan)
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = [tstart]
+
+    # initialzation of y[n], which will have variable length as well
+    y = Array(Ty, 1)
+    # TODO: fix after https://github.com/JuliaLang/julia/issues/16667
+    # y[1] = deepcopy(y0)
+    y[1] = copy(y0)
+
+    # initialzation of dy[n], which will have variable length as well
+    _dy = F(t[1], y[1])
+    Tdy = typeof(_dy)
+    dy = Tdy[_dy]
+
+    # Declare work variables for t_np1, y_np1 and dy_np1  which are for
+    # for t_{n+1}, y(t_{n+1}), and dy(t_{n+1})
+    t_np1   = tspan[1]
+    y_np1   = similar(y0, Eyf, dof)
+    dy_np1  = similar(dy[1])
+
+    # variables for dense output
+    tout = [tstart]
+    iter = 2
+    yout = Array(Ty, 1)
+    yout[1] = copy(y0)
+
+    #initialization of step size dt, which will have variable length as well
+    dt = initstep # current trial stepsize for a give step
+    prev_dt = dt # the stepsize of the last successful step
+    next_dt = 0.0 # stepsize to be used in next step
+
+    if dt == 0.
+        # initial guess at a step size
+        # dt, tdir, dy0 = hinit(F, y0, tstart, tfinal, 3, reltol, abstol)
+        dt, tdir, dy0 = hinit(F, y0, tstart, tfinal, 1, reltol, abstol)
+
+    else
+        tdir = sign(tfinal - tstart)
+    end
+
+    n = 1     #`n` counts the time steps. We start at n=1, the first step
+    steporder=1 # `steporder` is the order of the AB method at the current step
```

Comment:
Yes, this is true. It was indeed a large source of off-by-one errors. Using `stageorder` in replace of `steporder` does seem like a good approach to reduce the potential confusion.